src/vm.rs

```diff
@@ -131,23 +166,28 @@ impl<'a, S: Storage, C: PevmChain> VmDb<'a, S, C> {
             to,
             to_hash,
             to_code_hash: None,
-            is_lazy: false,
+            lazy_strategy: LazyStrategy::None,
             // Unless it is a raw transfer that is lazy updated, we'll
             // read at least from the sender and recipient accounts.
             read_set: ReadSet::with_capacity(2),
             read_accounts: HashMap::with_capacity_and_hasher(2, BuildIdentityHasher::default()),
         };
-        // We only lazy update raw transfers that already have the sender
-        // or recipient in [MvMemory] since sequentially evaluating memory
-        // locations with only one entry is much costlier than fully
-        // evaluating it concurrently.
         // TODO: Only lazy update in block syncing mode, not for block
         // building.
         if let Some(to) = to {
             db.to_code_hash = db.get_code_hash(*to)?;
-            db.is_lazy = db.to_code_hash.is_none()
-                && (vm.mv_memory.data.contains_key(&from_hash)
-                    || vm.mv_memory.data.contains_key(&to_hash.unwrap()));
+            db.lazy_strategy = LazyStrategy::from(from, &db.to_code_hash, &vm.txs[*tx_idx].data);
```

Comment:
At this point we can just pass in `&tx` to cover `nonce`, from`, `to`, and `input`?